tools/build_api.py

```diff
@@ -514,10 +515,29 @@ def build_project(src_paths, build_path, target, toolchain_name,
 
         # Link Program
         if toolchain.config.has_regions:
-            res, _ = toolchain.link_program(resources, build_path, name + "_application")
-            region_list = list(toolchain.config.regions)
-            region_list = [r._replace(filename=res) if r.active else r
-                           for r in region_list]
+            # create application binary
+            app_fn, _ = toolchain.link_program(resources, build_path, name + "_application")
+
+            # update region_list
+            region_list = list()
+            for region in toolchain.config.regions:
+                if region.name == "application":
+                    region = region._replace(filename = app_fn)
+                elif ((region.name == "firmware_metadata_header") and
+                      (not toolchain.config.building_bootloader)):
+                    # create metadata header binary
+                    app_bin_str = intelhex_offset(app_fn, 0).tobinstr()
+                    firmware_metadata_header = create_header(app_bin_str)
```

Comment:
Could we replace this with execution of a script outside of the scope of the tools? We might need to pass it some more arguments.